src/main/java/org/snomed/snowstorm/core/data/services/servicehook/CommitServiceHookClient.java

```diff
@@ -54,6 +55,15 @@ public void preCommitCompletion(Commit commit) throws IllegalStateException {
 			logger.error("Commit service hook failed for branch {}, commit {}, url {}, cookie {}",
 					commit.getBranch().getPath(), commit.getTimepoint().getTime(), serviceUrl,
 					obfuscateToken(authenticationToken), e);
+			// e will be HttpClientErrorException.Conflict if review for branch is incomplete.
+			// Regardless, if the commit is a promotion, we will always abort if an exception is encountered.
+			boolean promotion = commit.getCommitType().equals(Commit.CommitType.PROMOTION);
+			if (promotion) {
+				logger.error("Review for branch is incomplete; promotion aborted.");
+				throw new RuntimeServiceException("Review for branch is incomplete; promotion aborted.");
```

Comment:
The promotion will be aborted if an exception is encountered whilst communicating with AAG.

Reply:
We don't know which acceptance criteria hasn't been met. We could make the message more accurate here, something like: "Promotion blocked, authoring acceptance criteria not met."

Reply:
At first I couldn't understand why your version was more accurate than mine, but I suppose it's because my "review" is very abstract whereas yours is specific with "criteria". 

I have made this change.